pixl_imaging/src/pixl_imaging/_processing.py

```diff
@@ -143,27 +145,122 @@ async def _add_project_to_study(
         )
 
 
-async def _find_study_in_vna_or_raise(orthanc_raw: Orthanc, study: ImagingStudy) -> str:
+async def _find_study_in_archives_or_raise(orthanc_raw: Orthanc, study: ImagingStudy) -> str:
     """
-    Query the VNA for the study using its UID, fallback to querying on MRN + accession number if
-    UID is not available, raise exception if it doesn't exist
+    Query primary and secondary archives for a study.
+
+    - Query the primary archive for the study using its UID. If UID is not available, query on
+      MRN and accession number.
+    - If not found in the primary archive, and it is a weekday between 8 pm and 8 am, query the
+      online secondary archive using the study UID. If UID is not available, query on
+      MRN + accession number.
+    - If not found in the online secondary archive, query the nearline secondary archive. If found
+      in the nearline, assume the study is in the primary archive and log it.
```

Comment:
I'd probably say that we should have a discard error again here, with the cause being that it its in the nearline

Reply:
oh yeah that makes more sense